src/nccl_ofi_rdma.c

```diff
@@ -1176,14 +1184,18 @@ static inline int handle_bounce_recv(nccl_ofi_rdma_msg_type_t msg_type, nccl_net
 	}
 
 	bounce_data = get_bounce_data(bounce_req);
-	bounce_data->recv_len = cq_entry->len;
+	bounce_data->recv_len = msg_len;
+	bounce_fl_item = bounce_data->bounce_fl_item;
+
+	nccl_ofi_rdma_msg_type_t msg_type =
+		eager ? NCCL_OFI_RDMA_MSG_EAGER : *(nccl_ofi_rdma_msg_type_t *)&bounce_fl_item->bounce_msg;
```

Comment:
nccl_ofi_rdma_msg_type_t msg_type =
-               eager ? NCCL_OFI_RDMA_MSG_EAGER : *(nccl_ofi_rdma_msg_type_t *)&bounce_fl_item->bounce_msg;
+               eager ? NCCL_OFI_RDMA_MSG_EAGER : *(nccl_ofi_rdma_msg_type_t *)&bounce_fl_item->bounce_msg[device->msg_prefix_size];

Reply:
One more instance of this that needs to be fixed: https://github.com/AmedeoSapio/aws-ofi-nccl/blob/7c9fcb13e4130c4120ecde1f301ed515d55faa1c/src/nccl_ofi_rdma.c#L4446